packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx

```diff
@@ -105,30 +105,51 @@ export default class ProviderView extends View {
    * @returns {Promise}   Folders/files in folder
    */
   async getFolder (id, name) {
+    const controller = new AbortController()
+    const cancelRequest = () => controller.abort()
+
+    this.plugin.uppy.on('dashboard:close-panel', cancelRequest)
+    this.plugin.uppy.on('cancel-all', cancelRequest)
     this.setLoading(true)
+
     try {
-      const res = await this.provider.list(id)
       const folders = []
       const files = []
+      let path = this.nextPagePath || id
+
+      while (path && !controller.signal.aborted) {
```

Comment:
Maybe it's best to have `this.provider.list` throw so all cancellations are treated the same?
```suggestion
      while (path) {
```